src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php

```diff
@@ -1349,13 +1351,22 @@ private function getServiceCall($id, Reference $reference = null)
      */
     private function camelize($id)
     {
+        if (isset($this->serviceIdToMethodNameMap[$id])) {
+            return $this->serviceIdToMethodNameMap[$id];
+        }
+
         $name = Container::camelize($id);
+        $methodName = $name = preg_replace('/[^a-zA-Z0-9_\x7f-\xff]/', '', $name);
+        $suffix = 1;
 
-        if (!preg_match('/^[a-zA-Z0-9_\x7f-\xff]+$/', $name)) {
-            throw new InvalidArgumentException(sprintf('Service id "%s" cannot be converted to a valid PHP method name.', $id));
+        while (isset($this->usedMethodNames[$methodName])) {
+            ++$suffix;
+            $methodName = $name.$suffix;
         }
+        $this->serviceIdToMethodNameMap[$id] = $methodName;
+        $this->usedMethodNames[$methodName] = true;
```

Comment:
method names must be considered as case insensitive in this array, as PHP treats them this way. Otherwise you will still generated conflicting names